client/emv/emv_tags.c

```diff
@@ -471,6 +472,50 @@ static void emv_tag_dump_cvr(const struct tlv *tlv, const struct emv_tag *tag, F
 	return;
 }
 
+// EMV Book 3
+static void emv_tag_dump_cid(const struct tlv *tlv, const struct emv_tag *tag, FILE *f, int level) {
+	if (!tlv || tlv->len < 1) {
+		PRINT_INDENT(level);
+		fprintf(f, "\tINVALID!\n");
+		return;
+	}
+	
+	PRINT_INDENT(level);
+	if ((tlv->value[0] & 0xC0) == 0x00)	fprintf(f, "\tAC1: AAC (Transaction declined)\n");
+	if ((tlv->value[0] & 0xC0) == 0x40)	fprintf(f, "\tAC1: TC (Transaction approved)\n");
+	if ((tlv->value[0] & 0xC0) == 0x80)	fprintf(f, "\tAC1: ARQC (Online authorisation requested)\n");
+	if ((tlv->value[0] & 0xC0) == 0xC0)	fprintf(f, "\tAC1: RFU\n");
+
+	if ((tlv->value[0] & 0x08) != 0x00) {
```

Comment:
why  "not zero" when you want  one...

Reply:
this one? i thought its not so good)
(tlv->value[0] & 0x08) == 0x08

Reply:
either  
if ( tlv->value[0] & 0x08)  
or the long with comparison.
if (( tlv->value[0] & 0x08)  == 0x08)

both are ok,   and why not defines/enum for the bit you want to compare since you use that in the rest of the code?  What I dislike the most, is inconsistency

Reply:
OK